### PR TITLE
 [Entry] General: Environment Variables

### LINKED ISSUE #948: [Entry] General: Environment Variables
### Reason/inspiration (optional)

Please write a new concept entry for **environment variables** under the General topic.

- Since this is a General entry, try to avoid language-specific information.
- But do include info about its relationship with Unix and Unix-like variants (Linux).
  - Go ahead and link out to the Docs entries for [Linux](https://www.codecademy.com/resources/docs/general/linux) and [Unix](https://www.codecademy.com/resources/docs/general/unix), as well.
- Include a list of the most common environment variables. (`PATH`, `HOME`, etc.)

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by gracekishino**
### Description

- Add a new file for a new concept entry to general for environment variables [content/general/concepts/environment-variables/environment-variables.md](https://github.com/Codecademy/docs/compare/main...gracekishino:docs:environment-variables?expand=1#diff-b33de31118f430085b39dc80ea1eb2d2b32b44d62bf3a81bd2f1659e7beb4852)
- Add command line links to [documentation/catalog-content.md](https://github.com/Codecademy/docs/compare/main...gracekishino:docs:environment-variables?expand=1#diff-c7507cbbff329697341471d75a53d934c0298c847e761d9fa1b799a36a27097a)

Closes #948

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/general/concepts/environment-variables/environment-variables.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  - 'paths/computer-science'
+---
+
+**Environment variables** consist of a name/value pair and are set outside of applications, but accessible to them from the environment in which they are run.
+
+Environment variables are often set via the [terminal](https://www.codecademy.com/resources/docs/general/terminal) or in a script, though how they are managed and retrieved varies depending on the operating system or environment. For example, they can be used to store system configuration data or private keys for applications.
+
+## Relationship with Unix and Unix-like Variants
+
+On [Unix](https://www.codecademy.com/resources/docs/general/unix) and Unix-like variants such as [Linux](https://www.codecademy.com/resources/docs/open-source/linux), environment variables allow for configuration of the system and applications. They can contain information such as the default language for the system, which default text editor to use and paths to executable files.
```
Comment (SSwiniarski) [inline content/general/concepts/environment-variables/environment-variables.md:23]:
```suggestion
On [Unix](https://www.codecademy.com/resources/docs/general/unix) and Unix-like variants such as [Linux](https://www.codecademy.com/resources/docs/open-source/linux), environment variables allow for configuration of the system and applications. They can contain information such as the default language for the system, which default text editor to use, and paths to executable files.   
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@gracekishino I did a first review and only had one comment.

**Comment by gracekishino** [comment]
@SSwiniarski thanks for reviewing! I committed the change.

**Review by SSwiniarski — APPROVED** [review]
@gracekishino Looks good for a second review.

**Comment by gracekishino** [comment]
@yangc95 I made changes so that the tests should hopefully pass now.

File: documentation/catalog-content.md [hunk 2]
```diff
@@ -30,6 +30,13 @@ Feel free to add suggestions for new slugs to the lists as part of your PR! Be s
 - 'paths/back-end-engineer-career-path'
 ```
 
+### Command Line
+
+```
+- "learn-the-command-line"
+- "paths/computer-science"
```
Comment (yangc95) [inline documentation/catalog-content.md:37]:
```suggestion
- 'learn-the-command-line'
- 'paths/computer-science'
```

Comment (gracekishino) [reply documentation/catalog-content.md]: @yangc95 thank you for the suggestion! I have updated the code.

File: content/general/concepts/environment-variables/environment-variables.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  - 'Bash/Shell'
+  - 'Command Line'
+  - 'Linux'
+  - 'Unix'
+CatalogContent:
+  - 'learn-the-command-line'
+  - 'paths/computer-science'
+---
+
+**Environment variables** consist of a name/value pair and are set outside of applications, but accessible to them from the environment in which they are run.
```
Comment (yangc95) [inline content/general/concepts/environment-variables/environment-variables.md:17]: Can we re-word the second half of the sentence by specifying what is accessible and what is "them." Perhaps create a second sentence.

File: content/general/concepts/environment-variables/environment-variables.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+Environment variables are often set via the [terminal](https://www.codecademy.com/resources/docs/general/terminal) or in a script, though how they are managed and retrieved varies depending on the operating system or environment. For example, they can be used to store system configuration data or private keys for applications.
```
Comment (yangc95) [inline content/general/concepts/environment-variables/environment-variables.md:19]:
```suggestion
Environment variables are often set via the [terminal](https://www.codecademy.com/resources/docs/general/terminal) or in a script. They are managed and retrieved depending on the operating system or environment. For example, they can be used to store system configuration data or private keys for applications.
```

File: content/general/concepts/environment-variables/environment-variables.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+## Relationship with Unix and Unix-like Variants
+
+On [Unix](https://www.codecademy.com/resources/docs/general/unix) and Unix-like variants such as [Linux](https://www.codecademy.com/resources/docs/open-source/linux), environment variables allow for configuration of the system and applications. They can contain information such as the default language for the system, which default text editor to use, and paths to executable files.
```
Comment (yangc95) [inline content/general/concepts/environment-variables/environment-variables.md:23]:
```suggestion
On [Unix](https://www.codecademy.com/resources/docs/general/unix) and Unix-like variants such as [Linux](https://www.codecademy.com/resources/docs/open-source/linux), environment variables allow for configuration of the system and its applications. They can contain information such as the default language for the system, which default text editor to use, and paths to executable files.
```

File: content/general/concepts/environment-variables/environment-variables.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+## Some Common Environment Variables
+
+| Variable | Description                                                             |
+| -------- | ----------------------------------------------------------------------- |
+| `PATH`   | Contains a list of directories that the terminal searches for commands. |
+| `HOME`   | The default path to the current user's home directory                   |
+| `USER`   | The current user name                                                   |
+| `LANG`   | The default locale                                                      |
```
Comment (yangc95) [inline content/general/concepts/environment-variables/environment-variables.md:32]:
```suggestion
| `HOME`   | The default path to the current user's home directory.                   |
| `USER`   | The current user's name.                                                   |
| `LANG`   | The default locale.                                                      |
```

**Review by yangc95 — COMMENTED** [review]
@gracekishino Thanks for making that change! I had a few more comments prior to merge 🙂

Did you also want to take a look at this @Dusch4593?

File: content/general/concepts/environment-variables/environment-variables.md (hunk 3 above)
Comment (gracekishino) [reply content/general/concepts/environment-variables/environment-variables.md]: @yangc95 thank you for reviewing! I reworded this sentence. Hope it is OK now?

File: content/general/concepts/environment-variables/environment-variables.md (hunk 5 above)
Comment (gracekishino) [reply content/general/concepts/environment-variables/environment-variables.md]: @yangc95 thank you for this comment, I updated and added in "its".

File: content/general/concepts/environment-variables/environment-variables.md (hunk 6 above)
Comment (gracekishino) [reply content/general/concepts/environment-variables/environment-variables.md]: @yangc95 thanks for this suggestion! I updated and also took out some whitespace to line up the table.

File: content/general/concepts/environment-variables/environment-variables.md (hunk 4 above)
Comment (gracekishino) [reply content/general/concepts/environment-variables/environment-variables.md]: @yangc95 thanks for this suggestion! I changed the code and slightly reworded the sentence in a way I thought was clearer, but I can change to the above suggestion exactly if you think that is better.

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
@gracekishino thanks for your patience in getting this entry added to Docs!
Live link: https://www.codecademy.com/resources/docs/general/environment-variables
Screen Shot 2022-12-16 at 12 20 40 PM

cc: @SSwiniarski

**Comment by gracekishino** [comment]
@yangc95 @SSwiniarski thank you so much for reviewing and for your suggestions and comments 🙏 
Very excited to have my first open source contribution go live! 🥰

**Comment by yangc95** [comment]
@gracekishino Happy Holidays! This entry was moved to https://www.codecademy.com/resources/docs/command-line/environment-variables

Would you mind making a small change here to `name/value` -> `name-value` because the `/` makes it seem like they're interchangeable rather than a pair. TY for this change ^^ Wishing you a happy new year!

**Comment by gracekishino** [comment]
@yangc95 I can do that :) Wishing you a happy new year too!

**Pull request merged**